src/products/views/ProductCreate/handlers.ts

```diff
@@ -69,12 +82,39 @@ export function createHandler(
   }) => Promise<MutationFetchResult<ProductDelete>>
 ) {
   return async (formData: ProductCreateData) => {
+    let errors: Array<AttributeErrorFragment | UploadErrorFragment> = [];
+
+    const uploadFilesResult = await Promise.all(
+      formData.attributesWithNewFileValue.map(fileAttribute =>
+        uploadFile({
+          file: fileAttribute.value
+        })
+      )
+    );
+
+    const attributesWithAddedNewFiles: AttributeValueInput[] = uploadFilesResult.reduce(
+      (attributesWithAddedFiles, uploadFileResult, index) => {
+        const attribute = formData.attributesWithNewFileValue[index];
+
+        errors = [...errors, ...uploadFileResult.data.fileUpload.uploadErrors];
+        return [
+          ...attributesWithAddedFiles,
+          {
+            file: uploadFileResult.data.fileUpload.uploadedFile.url,
+            id: attribute.id,
+            values: []
+          }
+        ];
+      },
+      []
+    );
```

Comment:
Can we move this somewhere else to increase readability?

Reply:
sure